### PR TITLE
Add jobs for testing kubernetes-security org pull requests

### PR DESCRIPTION
**Pull request by jessfraz**
Things to do:

- [x] Add webhook to the repo itself, I was going to give @spxtr access rights.. but it looks like we need to add more seats to the org so idk if you can just email me what should be added for the hook in the meantime.
- [x] Add the service account file to the jenkins credentials so it matches `credentials.yaml` 
- [x] Add an ssh key with access to the repo to the jenkins instance for it to use for cloning
- [x] Give k8s-ci-robot access to the repo (requires buying another seat as well)

This only changes kubernetes jenkins pulls, I figured I would add the master/branch jobs in another step.

### CODE REVIEW

**Comment by fejta** [comment]
SO SORRY for missing this!!

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        job-name: pull-kubernetes-kubemark-e2e-gce-gci
+        json: 0
+        repo-name: 'github.com/kubernetes-security/kubernetes'
+        timeout: 0
+    - kubernetes-node-e2e:   # owner: jessfraz@google.com
+        max-total: 12
+        job-name: pull-kubernetes-node-e2e
+        json: 0
+        repo-name: 'github.com/kubernetes-security/kubernetes'
+        timeout: 0
```
Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml]: Please update json and timeout again, according to https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml#L164-L167, other jobs as well

**Comment by krzyzacy** [comment]
You can run `bazel test //...` locally to see if it'll pass :-)

**Comment by jessfraz** [comment]
@krzyzacy ya im debugging locally :)

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -1262,6 +1263,25 @@ def Check(job, name):
             'job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml',
             Check, use_json=is_modern)
 
+    def testBootstrapSecurityPullYaml(self):
+        is_modern = lambda name: 'unit' in name or 'verify' in name
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: This is also changed recently - https://github.com/kubernetes/test-infra/blob/master/jenkins/bootstrap_test.py#L1246-L1247

**Comment by krzyzacy** [comment]
LG for the bootstrap part, let @spxtr look at the prow side of things

**Comment by spxtr** [comment]
I think the idea is fine but I'm not sure I like the idea of copying the jobs like this. At the very least we should add a unit test that says that the k/k jobs need to exactly match the k-s/k jobs.

**Comment by jessfraz** [comment]
oh good call! and yeah it felt wrong to copy paste but i also didn't know
how to abstract further, I will add a test to make sure they match though!

**Comment by spxtr** [comment]
Alternatively and maybe less-grossly, we can add a special-case for this repo in https://github.com/kubernetes/test-infra/blob/master/prow/jobs/jobs.go#L203 .

File: prow/jobs/jobs.go [hunk 3]
```diff
@@ -205,6 +205,10 @@ func (ja *JobAgent) MatchingPresubmits(fullRepoName, body string, testAll *regex
 	defer ja.mut.Unlock()
 	var result []Presubmit
 	ott := testAll.MatchString(body)
+	// copy the jobs from k8s.io/kubernetes to kubernetes-security/kubernetes
+	if fullRepoName == "kubernetes-security/kubernetes" {
+		fullRepoName = "k8s.io/kubernetes"
```
Comment (spxtr) [inline prow/jobs/jobs.go]: I think it's `kubernetes/kubernetes`.

**Comment by spxtr** [comment]
The prow change looks good. Note that right now the jobs and logs will still show up on https://prow.k8s.io . Not sure if that's okay.

**Comment by jessfraz** [comment]
Ah yeah thanks for pointing that out, is there a certain way you would want
to implement a feature to have some jobs hidden on prow? Maybe for the time
being we could hide them, then implement something like auth. Because the
cloud bucket for artifacts we are going to have private as well. I don't
mind doing the implementation.

**Comment by spxtr** [comment]
Prow doesn't really have any concept of hiding things. It might actually be simpler to run a separate instance of prow entirely, or maybe run things in a different namespace is sufficient.

**Comment by jessfraz** [comment]
Added you all to that repo and I will rebase this

**Comment by spxtr** [comment]
The prow change lgtm. We will need to make some way of hiding these jobs from https://prow.k8s.io, but we can do that in a future PR.

**Comment by jessfraz** [comment]
sweet!

**Comment by spxtr** [comment]
Is there some way to guarantee that the Jenkins jobs stay up to date?

**Comment by liggitt** [comment]
are there any blockers to finishing up the items in this PR?

**Comment by spxtr** [comment]
My only worry is that `bootstrap-security-pull.yaml` will go out of sync with the other.

**Comment by jessfraz** [comment]
I can write something in python tests.

**Comment by jessfraz** [comment]
added a test

File: jenkins/bootstrap_test.py [hunk 4]
```diff
@@ -1311,6 +1311,18 @@ def Check(job, name):
             'job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml',
             Check, use_json=is_modern)
 
+    def testBootstrapSecurityPullYamlJobsMatch(self):
+        jobs2 = self.LoadBootstrapYaml('job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml')
+        for name, job in self.LoadBootstrapYaml('job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml').iteritems():
+            if job['repo-name'] is 'k8s.io/kubernetes':
```
Comment (spxtr) [inline jenkins/bootstrap_test.py]: `==` to compare strings, I think

Comment (jessfraz) [inline jenkins/bootstrap_test.py]: wow #GoLyfe tying to python... my bad

File: jenkins/bootstrap_test.py [hunk 5]
```diff
@@ -1311,6 +1311,18 @@ def Check(job, name):
...
+                job2 = jobs2[name]
+                self.assertEquals(job['max-total'], job2['max-total'])
```
Comment (spxtr) [inline jenkins/bootstrap_test.py]: is it possible to do a `for attr in job: self.assertEquals(job[attr], job2[attr])` or something along those lines? That way we don't need to manually list them.

Comment (jessfraz) [inline jenkins/bootstrap_test.py]: but the repo will need to be excluded, hmm i think there is a way to do that in a one-liner as well

Comment (spxtr) [inline jenkins/bootstrap_test.py]: It doesn't need to be a one-liner, I just don't want someone to be able to add an attribute to one file without the test catching it.

Comment (jessfraz) [inline jenkins/bootstrap_test.py]: fixed

**Comment by spxtr** [comment]
OK
/lgtm

**Comment by philips** [comment]
@jessfraz stuck on a conflict :(

**Comment by jessfraz** [comment]
will rebase

**Comment by jessfraz** [comment]
rebased!
@spxtr what permissions do I need for jenkins to add the ssh key/google cloud cred file? or can we get together for 30 minutes sometime and do it together?

**Comment by spxtr** [comment]
@fejta is the Jenkins master right now, I think you can ping him to give you temporary access to set that up.

**Comment by fejta** [comment]
I should have Jess unblocked, please to me if you need anything more.

**Comment by philips** [comment]
Thanks everyone!

File: jenkins/bootstrap.py [hunk 6]
```diff
@@ -684,6 +684,11 @@ def setup_magic_environment(job):
         'JENKINS_AWS_SSH_PUBLIC_KEY_FILE',
         os.path.join(home, '.ssh/kube_aws_rsa.pub'),
     )
+    os.environ.setdefault(
+        'K8S_SECURITY_SSH_PRIVATE_KEY_FILE',
```
Comment (jessfraz) [inline jenkins/bootstrap.py]: I added this field with the ssh key and the credential to jenkins but I am unsure if we want to inject this for all jobs, seems weird but I'm not sure where else to put it to make sure it only goes on that specific repos jobs

**Comment by krzyzacy** [comment]
@rmmh https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/1515/pull-test-infra-bazel/1519/ this is the flake I was talking about :-)

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

File: jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    - string:
+        name: PULL_BASE_REF
+    # The test job tracks a run through the queue using the buildId parameter.
+    - string:
+        name: buildId
+    wrappers:
+    - e2e-credentials-binding
+    - credentials-binding:
+      - file:
+        credential-id: 'k8s-security-gcp-service-account'
```
Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-security-pull.yaml]: indent in, I think

**Comment by jessfraz** [comment]
updated!

**Comment by krzyzacy** [comment]
should be fine, feel free to try it out

**Comment by jessfraz** [comment]
how do i try it, sorry I'm a newb

**Comment by krzyzacy** [comment]
eh, I mean, merge and watch it burn?

**Comment by jessfraz** [comment]
ok im nervous, that's what I was thinking haha

**Pull request merged**